controller/setting_controller.go

```diff
@@ -323,11 +323,20 @@ func (sc *SettingController) syncDangerZoneSettingsForManagedComponents(settingN
 				return err
 			}
 		case types.SettingNameStorageNetwork:
-			fnAreAllVolumeDetached := func() (bool, error) {
-				isDetached, _, err := sc.ds.AreAllVolumesDetached(longhorn.DataEngineTypeAll)
-				return isDetached, err
+			funcPreupdate := func() error {
+				detached, _, err := sc.ds.AreAllVolumesDetached(longhorn.DataEngineTypeAll)
+				if err != nil {
+					return errors.Wrapf(err, "failed to check volume detachment for %v setting update", types.SettingNameStorageNetwork)
+				}
+
+				if !detached {
+					return &types.ErrorInvalidState{Reason: fmt.Sprintf("failed to apply %v setting to Longhorn components when there are attached volumes. It will be eventually applied", types.SettingNameStorageNetwork)}
```

Comment:
nit: We can generalize a package for errors with different type such InvalidState, NotFound and so on.